docs/source/testing/integration.rst

```diff
@@ -18,4 +18,135 @@
+
+The following code-block provides an example for how one can add the option for
+an integration test to a project that uses the NWX ecosystem.
+
+.. code-block:: CMake
+
+    # Include option
+    cmaize_option_list(
+        BUILD_INTEGRATION_TESTS ON "Build the integration tests?"
+    )
+
+    # How to build the integration tests
+    if("${BUILD_INTEGRATION_TESTS}")
+        include(nwx_pybind11)
+        # Set relevant test directories
+        set(CXX_INCLUDE_DIR /path/to/cxx/includes)
+        set(CXX_TEST_DIR /path/to/cxx/integration/tests)
+        set(PYTHON_TEST_DIR /path/to/python/integration/tests)
+
+        # Build NWChemEx for the test
+        cmaize_find_or_build_dependency(
+            nwchemex
+            URL github.com/NWChemEx/NWChemEx
+            VERSION master
+            BUILD_TARGET nwchemex
+            FIND_TARGET nwx::nwchemex
+            CMAKE_ARGS BUILD_TESTING=OFF
+                       BUILD_PYBIND11_PYBINDINGS=ON
+        )
+
+        # Add integration tests
+        nwx_pybind11_tests(
+            py_test_integration_scf
+            "${PYTHON_TEST_DIR}/test_main.py"
+            SUBMODULES parallelzone pluginplay chemist simde chemcache friendzone nwchemex
+        )
+    endif()
+
+Integration Tests
+=================
+
+Building on the description provided in :ref:`writing_unit_tests`, integration
+tests are written in the same manner. Below is an example of how to use NWChemEx
+in our new integration test to acquire input values and submodules that may be
+needed by a module in our project.
+
+.. code-block:: python
+
+    import unittest
+    import nwchemex
+    import scf
+    from pluginplay import ModuleManager
+    from simde import AOEnergy
+    from simde import MoleculeFromString
+    from simde import MolecularBasisSet
+
+    class TestIntegration(unittest.TestCase):
+
+        def test_scf_module(self):
+            # Module we want to test
+            key = "SCF Module"
+
+            # Property Types from SimDE
+            molecule_pt = MoleculeFromString()
+            basis_set_pt = MolecularBasisSet()
+            energy_pt = AOEnergy()
+
+            # Can use NWChemEx modules to get inputs
+            mol = self.mm.run_as(molecule_pt, "NWX Molecules", "water")
+            bs = self.mm.run_as(basis_set_pt, "sto-3g", mol)
+
+            # set NWChemEx modules as needed submodules
+            submod_key = "A submodule of my SCF module"
+            integral_key = "Some integral needed to run SCF"
+            mm.change_submod(key, submod_key, integral_key)
+
+            # Test our module
+            egy = self.mm.run_as(energy_pt, key, mol, bs)
+            self.assertAlmostEqual(egy, 3.14159265359, places=6)
+
+        def test_another_module(self):
+            # Add more tests where appropriate
+            pass
+
+        def setUp(self):
+            self.mm = ModuleManager()
+            nwchemex.load_modules(mm) # Also loads out SCF modules
```

Comment:
Fine for now, but not sure this is going to work in practice. In particular, I'm not convinced it will find the SCF version we want to test vs. the release version NWX depends on. We can worry about that later though.

Reply:
I wondered about that, as well. My expectation is that the version that is being built at the time should be found by CMake, so it shouldn't try to pull and build the version specified within NWChemEx. I think it should also take precedent over installed versions, but I'm not sure.

Reply:
I think it's going to depend on the Python path order if there's an installed version competing with the just built version. An "easy" way around this is:

```.py
import scf # Ensure import of this repo
import nwchemex

self.mm = ModuleManager()
scf.load_modules(mm)
nwchemex.load_modules(mm)
```

The trick would be to make sure that `load_modules` is a no-op if the keys already exist.

Reply:
It's also worth noting this only an issue for NWChemEx authors. Contributors don't have their plugins shipped with NWChemEx.